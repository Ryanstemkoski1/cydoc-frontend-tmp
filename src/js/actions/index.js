import {ADD_NOTE, ADD_TEMPLATE, DATA_LOADED, LOGIN_REQUEST, LOGOUT, SAVE_NOTE} from "../constants/action-types";
import axios from 'axios'
import api from "../constants/api";

export function addNote(payload) {
    return { type: ADD_NOTE, payload };
}

export function addTemplate(payload) {
    return { type: ADD_TEMPLATE, payload: payload}
}

export function getAllRecords() {
    return function(dispatch) {
        return axios.get(api.records.dev)
            .then(response => response.data)
            .then(data => {
                dispatch({ type: DATA_LOADED, payload: data });
            });
    };
}

export function loginRequest(payload){
    return function(dispatch) {
        return axios.post(api.login.dev, payload)
            .then(res => {
                const user = res;
                console.log(user)
                localStorage.setItem('user', JSON.stringify(user));
                return user;
            })
            .then((user)=> {
                dispatch({type: LOGIN_REQUEST, payload: user.data})
                return user;
            })
            .catch(err => {
                return err.response;
            })
    }
}

export function logout(){
        localStorage.removeItem('user');
        return { type: LOGOUT };
}

export  function saveNote(payload){
    return function (dispatch) {
        const headers = {
            'Content-Type': 'application/json',

        };
        return axios.post(api.newRecord.dev,
            { headers: {
                'Content-Type': 'application/json'}},
            payload)
            .then(res => res.data)
            .then(data => {
                dispatch({type: SAVE_NOTE, payload: data})
            })
    }
}