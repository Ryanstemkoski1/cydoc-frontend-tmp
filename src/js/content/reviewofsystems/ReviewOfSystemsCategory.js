import React, {Component, Fragment} from 'react'
import {Grid, Form, Checkbox} from 'semantic-ui-react'
import PropTypes from 'prop-types'

//Sub-Component that represents the individual categories for the Review of Systems section of the note
export default class ReviewOfSystemsCategory extends Component{

    constructor(props) {
        super(props)

        this.category = this.props.category
        this.options = this.props.options
        this.state = {}
    }

    handleChange = (option, value) => {
        this.setState({[option]: value})
    }

    render() {
        return (
            <Grid.Column>
                <h3>{this.category}</h3>
                <Form>
                    <label>Y N</label>
                    {this.options.map(
                    option => 
                        <Form.Group inline>
                            <Checkbox radio value='y' checked={this.state[option] === 'y'} onChange={(e, {value}) => this.handleChange(option, value)}/>
                            <Checkbox radio value='n' checked={this.state[option] === 'n'} onChange={(e, {value}) => this.handleChange(option, value)}/>
                            <label style={{paddingLeft: "10px"}}>{option}</label>
                        </Form.Group>
                    )}
                </Form>
            </Grid.Column>
        );
    }

    
}

ReviewOfSystemsCategory.propTypes = {
    category: PropTypes.string.isRequired, //Heading that goes over the divider
    options: PropTypes.array.isRequired, //Options to choose from the category
}