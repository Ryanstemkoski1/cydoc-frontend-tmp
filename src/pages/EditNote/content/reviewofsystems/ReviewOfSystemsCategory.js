import React, {Component} from 'react';
import {Grid, Button, Card, Container, Segment, Header} from 'semantic-ui-react';
import PropTypes from 'prop-types';
import HPIContext from 'contexts/HPIContext.js';
import './ReviewOfSystems.css';

//Sub-Component that represents the individual categories for the Review of Systems section of the note
export default class ReviewOfSystemsCategory extends Component{

    static contextType = HPIContext

    constructor(props) {
        super(props)

        this.category = this.props.category
        this.options = this.props.options
        this.state = {}
    }

    handleChange = (option, value) => {
        const values = this.context["Review of Systems"]
        values[this.category][option] = value
        this.context.onContextChange("Review of Systems", values)
    }

    render() {
        return (
            <Segment raised>
                <Header>{this.category}</Header>
                    <Grid>
                    {this.options.map((option) => 
                        <Grid.Row key={option}>
                            <Grid.Column width={4} className="no-padding">
                                <Button
                                    compact
                                    floated='right'
                                    color={this.context["Review of Systems"][this.category][option] === 'n' ? 'green' : null}
                                    value='n'
                                    active={this.context["Review of Systems"][this.category][option] === 'n'}
                                    onClick={(e, {value}) => this.handleChange(option, value)}
                                >
                                    NO
                                </Button>
                            </Grid.Column>
                            <Grid.Column
                                width={7}
                                verticalAlign='middle'
                                className="ros-symptom no-padding"
                            >
                                {option}
                            </Grid.Column>
                            <Grid.Column width={4} className="no-padding">
                                <Button
                                    compact
                                    floated='left'
                                    color={this.context["Review of Systems"][this.category][option] === 'y' ? 'red' : null}
                                    value='y' active={this.context["Review of Systems"][this.category][option] === 'y'}
                                    onClick={(e, {value}) => this.handleChange(option, value)} 
                                >
                                    YES
                                </Button>
                            </Grid.Column>
                        </Grid.Row>
                    )}
                    </Grid>
            </Segment>
        );
    }
}

ReviewOfSystemsCategory.propTypes = {
    category: PropTypes.string.isRequired, //Heading that goes over the divider
    options: PropTypes.array.isRequired, //Options to choose from the category
}