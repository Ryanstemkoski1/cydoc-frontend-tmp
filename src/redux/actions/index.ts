import { ROSActionTypes } from './reviewOfSystemsActions';
import { MedicalHistoryActionTypes } from './medicalHistoryActions';
import { SurgicalHistoryActionTypes } from './surgicalHistoryActions';
import { MedicationsActionTypes } from './medicationsActions';
import { AllergiesActionTypes } from './allergiesActions';
import { SocialHistoryActionTypes } from './socialHistoryActions';
import { FamilyHistoryActionTypes } from './familyHistoryActions';
import { PhysicalExamActionTypes } from './physicalExamActions';
import { PlanActionTypes } from './planActions';
import { HpiActionTypes } from './hpiActions';
import { WidgetActionTypes } from './widgetActions';
import { chiefComplaintsActionTypes } from './chiefComplaintsActions';

export type AllActionTypes =
    | ROSActionTypes
    | PhysicalExamActionTypes
    | WidgetActionTypes
    | MedicalHistoryActionTypes
    | SurgicalHistoryActionTypes
    | MedicationsActionTypes
    | AllergiesActionTypes
    | SocialHistoryActionTypes
    | FamilyHistoryActionTypes
    | PlanActionTypes
    | HpiActionTypes
    | chiefComplaintsActionTypes;
