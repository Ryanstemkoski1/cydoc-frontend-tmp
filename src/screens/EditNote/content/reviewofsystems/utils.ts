import { YesNoResponse } from '@constants/enums';

export const initialStore = {
    Eyes: {
        Glasses: YesNoResponse.None,
        Contacts: YesNoResponse.None,
        Blurriness: YesNoResponse.None,
    },
    Throat: {
        Lumps: YesNoResponse.None,
        Swelling: YesNoResponse.None,
        Pain: YesNoResponse.None,
        Stiffness: YesNoResponse.None,
    },
    Nose: {
        Colds: YesNoResponse.None,
        Stuffiness: YesNoResponse.None,
        Sneezing: YesNoResponse.None,
        Itching: YesNoResponse.None,
        Allergy: YesNoResponse.None,
        Nosebleed: YesNoResponse.None,
    },
};
