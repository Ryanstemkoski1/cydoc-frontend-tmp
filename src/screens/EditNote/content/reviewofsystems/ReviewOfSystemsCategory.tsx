import YesAndNo from '@components/tools/YesAndNo/YesAndNo';
import { YesNoResponse } from '@constants/enums';
import React, { Component } from 'react';
import { connect, ConnectedProps } from 'react-redux';
import { selectManyHandleClick } from '@redux/actions/hpiActions';
import { toggleROSOption } from '@redux/actions/reviewOfSystemsActions';
import { CurrentNoteState } from '@redux/reducers';
import { ReviewOfSystemsState } from '@redux/reducers/reviewOfSystemsReducer';
import {
    selectReviewOfSystemsOptions,
    selectReviewOfSystemsState,
} from '@redux/selectors/reviewOfSystemsSelectors';
import { Header } from 'semantic-ui-react';
import AllNegativeButton from './AllNegativeButton.js';
import './ReviewOfSystems.css';
import style from './ReviewOfSystemsCategory.module.scss';
import HandleWriteInInput from '../hpi/knowledgegraph/components/responseComponents/HandleWriteInInput';

interface OwnProps {
    category: string;
    key: string;
    node: string;
    selectManyState: ReviewOfSystemsState;
    selectManyOptions: string[];
}

type ReduxProps = ConnectedProps<typeof connector>;

interface State {
    ROSState: ReviewOfSystemsState;
}

type Props = ReduxProps & OwnProps;

class ReviewOfSystemsCategory extends Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = { ROSState: this.props.ROSState };
    }

    componentDidUpdate(prevProps: Props) {
        if (prevProps.ROSState !== this.props.ROSState) {
            this.setState({ ROSState: this.props.ROSState });
        }
    }

    handleChange = (option: string, value: YesNoResponse) => {
        if (!this.isSelectMany()) {
            this.props.toggleROSOption(this.props.category, option, value);
            this.setState({ ROSState: this.props.ROSState });
        } else {
            this.props.selectManyHandleClick(this.props.node, option, value);
            this.setState((prevState) => ({
                ROSState: {
                    ...prevState.ROSState,
                    ['']: {
                        ...prevState.ROSState[''],
                        [option]:
                            value === prevState.ROSState[''][option]
                                ? ''
                                : value,
                    },
                },
            }));
        }
    };

    isSelectMany = () => {
        return Object.keys(this.props.selectManyState).length != 0;
    };

    breakWord = (category: string): string | undefined => {
        const slash = '/';
        if (category.includes(slash)) {
            return category.split('/').join(' / ');
        } else {
            return category;
        }
    };

    render() {
        const { category, ROSOptions } = this.props;
        let { ROSState } = this.props;
        if (this.isSelectMany()) {
            ROSState = this.state.ROSState;
        }
        return (
            <div className={style.reviewOfSystems}>
                {!this.isSelectMany() && (
                    <Header
                        as='h3'
                        className={`header-titles ${style.reviewOfSystems__caption}`}
                    >
                        {this.breakWord(category)}
                    </Header>
                )}

                <AllNegativeButton handleClick={this.handleChange}>
                    {ROSOptions.map((option: string) => (
                        <div
                            className={`${style.reviewOfSystems__item} reviewOfSystemsItem flex flex-wrap align-center justify-between`}
                            key={option}
                        >
                            <YesAndNo
                                yesButtonActive={
                                    ROSState[category][option] ===
                                    YesNoResponse.Yes
                                }
                                handleYesButtonClick={() =>
                                    this.handleChange(option, YesNoResponse.Yes)
                                }
                                noButtonActive={
                                    ROSState[category][option] ===
                                    YesNoResponse.No
                                }
                                handleNoButtonClick={() => {
                                    this.handleChange(option, YesNoResponse.No);
                                }}
                            />
                            {option.toLowerCase() === 'other' &&
                            ROSState[category][option] === YesNoResponse.Yes ? (
                                <HandleWriteInInput
                                    name={option}
                                    node={this.props.node}
                                    options={ROSOptions}
                                />
                            ) : (
                                <p>{option.replace('Î”', 'Changes in')}</p>
                            )}
                            <YesAndNo
                                yesButtonActive={
                                    ROSState[category][option] ===
                                    YesNoResponse.Yes
                                }
                                handleYesButtonClick={() =>
                                    this.handleChange(option, YesNoResponse.Yes)
                                }
                                noButtonActive={
                                    ROSState[category][option] ===
                                    YesNoResponse.No
                                }
                                handleNoButtonClick={() => {
                                    this.handleChange(option, YesNoResponse.No);
                                }}
                            />
                        </div>
                    ))}
                </AllNegativeButton>
            </div>
        );
    }
}

const mapStateToProps = (state: CurrentNoteState, ownProps: OwnProps) => ({
    ROSState:
        Object.keys(ownProps.selectManyState).length == 0
            ? selectReviewOfSystemsState(state)
            : ownProps.selectManyState,
    ROSOptions:
        Object.keys(ownProps.selectManyState).length == 0
            ? selectReviewOfSystemsOptions(state, ownProps.category)
            : ownProps.selectManyOptions,
});

const mapDispatchToProps = {
    toggleROSOption: toggleROSOption,
    selectManyHandleClick: selectManyHandleClick,
};

const connector = connect(mapStateToProps, mapDispatchToProps);

export default connector(ReviewOfSystemsCategory);
