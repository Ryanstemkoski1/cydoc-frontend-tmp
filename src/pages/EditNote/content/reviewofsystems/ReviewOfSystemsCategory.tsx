import YesAndNo from 'components/tools/YesAndNo/YesAndNo';
import { YesNoResponse } from 'constants/enums';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
    selectManyHandleClick,
    SelectManyHandleClickAction,
} from 'redux/actions/hpiActions';
import {
    toggleROSOption,
    ToggleROSOptionAction,
} from 'redux/actions/reviewOfSystemsActions';
import { CurrentNoteState } from 'redux/reducers';
import { ReviewOfSystemsState } from 'redux/reducers/reviewOfSystemsReducer';
import {
    selectReviewOfSystemsOptions,
    selectReviewOfSystemsState,
} from 'redux/selectors/reviewOfSystemsSelectors';
import { Header } from 'semantic-ui-react';
import AllNegativeButton from './AllNegativeButton.js';
import './ReviewOfSystems.css';
import style from './ReviewOfSystemsCategory.module.scss';

interface CategoryProps {
    key: string;
    category: string;
    selectManyState: ReviewOfSystemsState;
    node: string;
}

interface StateProps {
    ROSState: ReviewOfSystemsState;
    ROSOptions: string[];
}

interface OwnProps {
    category: string;
    selectManyState: ReviewOfSystemsState;
    selectManyOptions: string[];
}

interface DispatchProps {
    toggleROSOption: (
        category: string,
        option: string,
        yesOrNo: YesNoResponse
    ) => ToggleROSOptionAction;
    selectManyHandleClick: (
        medId: string,
        option: string,
        yesOrNo: YesNoResponse
    ) => SelectManyHandleClickAction;
}

interface OwnState {
    ROSState: ReviewOfSystemsState;
}

const mapStateToProps = (
    state: CurrentNoteState,
    ownProps: OwnProps
): StateProps => ({
    ROSState:
        Object.keys(ownProps.selectManyState).length == 0
            ? selectReviewOfSystemsState(state)
            : ownProps.selectManyState,
    ROSOptions:
        Object.keys(ownProps.selectManyState).length == 0
            ? selectReviewOfSystemsOptions(state, ownProps.category)
            : ownProps.selectManyOptions,
});

const mapDispatchToProps = {
    toggleROSOption: toggleROSOption,
    selectManyHandleClick: selectManyHandleClick,
};

type Props = CategoryProps & DispatchProps & StateProps;
type State = OwnState;

class ReviewOfSystemsCategory extends Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = { ROSState: this.props.ROSState };
    }

    componentDidUpdate(prevProps: Props) {
        if (prevProps.ROSState !== this.props.ROSState) {
            this.setState({ ROSState: this.props.ROSState });
        }
    }

    handleChange = (option: string, value: YesNoResponse) => {
        if (!this.isSelectMany()) {
            this.props.toggleROSOption(this.props.category, option, value);
            this.setState({ ROSState: this.props.ROSState });
        } else {
            this.props.selectManyHandleClick(this.props.node, option, value);
            this.setState((prevState) => ({
                ROSState: {
                    ...prevState.ROSState,
                    ['']: {
                        ...prevState.ROSState[''],
                        [option]:
                            value === prevState.ROSState[''][option]
                                ? ''
                                : value,
                    },
                },
            }));
        }
    };

    isSelectMany = () => {
        return Object.keys(this.props.selectManyState).length != 0;
    };

    breakWord = (category: string): string | undefined => {
        let header = '';
        const slash = '/';
        if (category.includes(slash)) {
            return (header = category.split('/').join(' / '));
        } else {
            return (header = category);
        }
    };

    render() {
        const { category, ROSOptions } = this.props;
        let { ROSState } = this.props;
        if (this.isSelectMany()) {
            ROSState = this.state.ROSState;
        }
        return (
            <div className={style.reviewOfSystems}>
                {!this.isSelectMany() && (
                    <Header
                        as='h3'
                        className={`header-titles ${style.reviewOfSystems__caption}`}
                    >
                        {this.breakWord(category)}
                    </Header>
                )}

                <AllNegativeButton handleClick={this.handleChange}>
                    {ROSOptions.map((option: string) => (
                        <div
                            className={`${style.reviewOfSystems__item} reviewOfSystemsItem flex align-center justify-between`}
                            key={option}
                        >
                            <YesAndNo
                                yesButtonActive={
                                    ROSState[category][option] ===
                                    YesNoResponse.Yes
                                }
                                handleYesButtonClick={() =>
                                    this.handleChange(option, YesNoResponse.Yes)
                                }
                                noButtonActive={
                                    ROSState[category][option] ===
                                    YesNoResponse.No
                                }
                                handleNoButtonClick={() => {
                                    this.handleChange(option, YesNoResponse.No);
                                }}
                            />
                            <p>{option.replace('Î”', 'Changes in')}</p>

                            <YesAndNo
                                yesButtonActive={
                                    ROSState[category][option] ===
                                    YesNoResponse.Yes
                                }
                                handleYesButtonClick={() =>
                                    this.handleChange(option, YesNoResponse.Yes)
                                }
                                noButtonActive={
                                    ROSState[category][option] ===
                                    YesNoResponse.No
                                }
                                handleNoButtonClick={() => {
                                    this.handleChange(option, YesNoResponse.No);
                                }}
                            />
                        </div>
                    ))}
                </AllNegativeButton>
            </div>
        );
    }
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ReviewOfSystemsCategory);
