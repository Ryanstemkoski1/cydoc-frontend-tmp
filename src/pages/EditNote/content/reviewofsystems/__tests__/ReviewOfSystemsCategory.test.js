import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import ReviewOfSystemsCategory from '../ReviewOfSystemsCategory';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { initialStore } from '../utils';
import { ROS_ACTION } from 'redux/actions/actionTypes';
import { Button } from 'semantic-ui-react';

Enzyme.configure({ adapter: new EnzymeAdapter() });

const mockStore = configureStore([]);

const mountWithStore = (
    initStore = { reviewOfSystems: initialStore },
    props = { category: 'Eyes' }
) => {
    const store = mockStore(initStore);
    return {
        store,
        wrapper: mount(
            <Provider store={store}>
                <ReviewOfSystemsCategory {...props} />
            </Provider>
        ),
    };
};

describe('ReviewOfSystemsCategory', () => {
    it('renders without crashing', () => {
        const { wrapper } = mountWithStore();
        expect(wrapper).toBeTruthy();
    });

    it('matches snapshot', () => {
        const { wrapper } = mountWithStore();
        expect(wrapper.html()).toMatchSnapshot();
    });

    // // TODO: Fix below tests
    // it('renders Yes No buttons', () => {
    //     const { wrapper } = mountWithStore();
    //     const button = wrapper.find('button');
    //     expect(button).toHaveLength(6);
    // });

    it('handles change when YES is clicked', () => {
        const { store, wrapper } = mountWithStore();
        const button = wrapper
            .find(Button)
            .find(`[aria-label='yes-button']`)
            .first();

        expect(button).toBeTruthy();
        button.simulate('click', {
            target: { value: 'YES' },
        });
        const expectedAction = [
            {
                type: ROS_ACTION.TOGGLE_OPTION,
                payload: {
                    category: Object.keys(initialStore)[0],
                    option: 'Glasses',
                    yesOrNo: 'YES',
                },
            },
        ];
        expect(store.getActions()).toEqual(expectedAction);
    });

    it('handles change when NO is clicked', () => {
        const { store, wrapper } = mountWithStore();
        const button = wrapper
            .find(Button)
            .find(`[aria-label='no-button']`)
            .first();

        expect(button).toBeTruthy();
        button.simulate('click', {
            target: { value: 'NO' },
        });
        const expectedAction = [
            {
                type: ROS_ACTION.TOGGLE_OPTION,
                payload: {
                    category: Object.keys(initialStore)[0],
                    option: 'Glasses',
                    yesOrNo: 'NO',
                },
            },
        ];
        expect(store.getActions()).toEqual(expectedAction);
    });
});
