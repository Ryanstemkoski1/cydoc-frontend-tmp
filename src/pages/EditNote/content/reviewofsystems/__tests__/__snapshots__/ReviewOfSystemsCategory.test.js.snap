// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewOfSystemsCategory matches snapshot 1`] = `"<div class=\\"reviewOfSystems\\"><h3 class=\\"ui header header-titles\\">Eyes</h3><div class=\\"symptomsBlock\\"><div class=\\"symptomsBlock__btn flex align-center\\"><p></p><button class=\\"symptomsBlock__all button outline info pill sm false\\" data-hover=\\"false\\">ALL NO</button></div><div class=\\"reviewOfSystems__item flex-wrap align-center\\"><p>Glasses</p><aside><div class=\\"YesNoButton  flex\\"><button title=\\"Yes\\" condition=\\"\\" data-hover=\\"false\\" class=\\"button outline info pill sm \\">Yes</button><button title=\\"No\\" condition=\\"\\" data-hover=\\"false\\" class=\\"button outline info pill sm \\">No</button></div></aside></div><div class=\\"reviewOfSystems__item flex-wrap align-center\\"><p>Contacts</p><aside><div class=\\"YesNoButton  flex\\"><button title=\\"Yes\\" condition=\\"\\" data-hover=\\"false\\" class=\\"button outline info pill sm \\">Yes</button><button title=\\"No\\" condition=\\"\\" data-hover=\\"false\\" class=\\"button outline info pill sm \\">No</button></div></aside></div><div class=\\"reviewOfSystems__item flex-wrap align-center\\"><p>Blurriness</p><aside><div class=\\"YesNoButton  flex\\"><button title=\\"Yes\\" condition=\\"\\" data-hover=\\"false\\" class=\\"button outline info pill sm \\">Yes</button><button title=\\"No\\" condition=\\"\\" data-hover=\\"false\\" class=\\"button outline info pill sm \\">No</button></div></aside></div></div></div>"`;
